model/test_results.go

```diff
@@ -1015,3 +1063,81 @@ func GetTestResultsStats(ctx context.Context, env cedar.Environment, opts FindTe
+	catcher.NewWhen(o.StartAt.After(o.EndAt), "the start date cannot be greater than the end date")
+	catcher.NewWhen(o.EndAt.IsZero(), "end date cannot be zero")
+
+	return catcher.Resolve()
+}
+
+// FindTestResultsByProject returns all test results in the database with the
+// given set of project names and within the given date interval.
+func FindTestResultsByProject(ctx context.Context, env cedar.Environment, opts FindTestResultsByProjectOptions) ([]TestResults, error) {
+	if err := opts.validate(); err != nil {
+		return nil, errors.Wrap(err, "invalid options")
+	}
+
+	filter := bson.M{
+		bsonutil.GetDottedKeyName(testResultsInfoKey, testResultsInfoProjectKey): bson.M{"$in": opts.Projects},
+		testResultsCreatedAtKey: bson.M{
+			"$gte": opts.StartAt,
+			"$lte": opts.EndAt,
+		},
+	}
+
+	cur, err := env.GetDB().Collection(testResultsCollection).Find(ctx, filter)
+	if err != nil {
+		return nil, errors.Wrap(err, "finding test results")
+	}
+
+	var results []TestResults
+	if err = cur.All(ctx, &results); err != nil {
+		return nil, errors.Wrap(err, "decoding test results")
+	}
+
+	for i := range results {
+		results[i].env = env
+		results[i].populated = true
+	}
+
+	return results, nil
+}
+
+// ParquetTestResults describes a set of test results from a task execution to
+// be stored in Apache Parquet format.
+type ParquetTestResults struct {
+	Version   string              `parquet:"name=version, type=BYTE_ARRAY, convertedtype=UTF8"`
+	Variant   string              `parquet:"name=variant, type=BYTE_ARRAY, convertedtype=UTF8"`
+	TaskID    string              `parquet:"name=task_id, type=BYTE_ARRAY, convertedtype=UTF8"`
+	Execution int32               `parquet:"name=execution, type=INT32"`
+	CreatedAt int64               `parquet:"name=created_at, type=INT64, logicaltype=TIMESTAMP, logicaltype.unit=MILLIS, logicaltype.isadjustedtoutc=true"`
+	Results   []ParquetTestResult `parquet:"name=results, type=LIST"`
+}
+
+// ParquetTestResult describes a single test result to be stored in Apache
+// Parquet file format.
+type ParquetTestResult struct {
+	TestName        string  `parquet:"name=test_name, type=BYTE_ARRAY, convertedtype=UTF8"`
+	DisplayTestName *string `parquet:"name=display_test_name, type=BYTE_ARRAY, convertedtype=UTF8"`
+	GroupID         *string `parquet:"name=group_id, type=BYTE_ARRAY, convertedtype=UTF8"`
+	Trial           *int32  `parquet:"name=trial, type=INT32"`
+	Status          string  `parquet:"name=status, type=BYTE_ARRAY, convertedtype=UTF8"`
+	LogTestName     *string `parquet:"name=log_test_name, type=BYTE_ARRAY, convertedtype=UTF8"`
+	LogURL          *string `parquet:"name=log_url, type=BYTE_ARRAY, convertedtype=UTF8"`
+	RawLogURL       *string `parquet:"name=raw_log_url, type=BYTE_ARRAY, convertedtype=UTF8"`
+	LineNum         *int32  `parquet:"name=line_num, type=INT32"`
+	TestStartTime   int64   `parquet:"name=test_start_time, type=INT64, logicaltype=TIMESTAMP, logicaltype.unit=MILLIS, logicaltype.isadjustedtoutc=true"`
```

Comment:
@rtimmons what do you think?

Reply:
I think it is fine to leave it as is since the struct tags make it explicit that this a millisecond unix timestamp in parquet. I am going to merge this.